src/shared/types/DomainHookReturnValue.ts

```diff
@@ -2,3 +2,9 @@ export type DomainHookReturnValue<T = any, V = any> = {
   data: Record<string, T>;
   actions: Record<string, V>;
 };
+
+// TODO: Christian is struggling with type safefty of my domain hook return values, is this a good idea?
```

Comment:
I had basically no safety with `DomainHookReturnValue`. Unsure if I am just not aware how to make the returns typed with it. But this saved me a lot of trouble